sched/signal/sig_dispatch.c

```diff
@@ -487,6 +486,17 @@ int nxsig_tcbdispatch(FAR struct tcb_s *stcb, siginfo_t *info,
 
   flags = nxsig_alloc_dyn_pending(flags);
 
+  /* If there are no available items for pending signals, and this was a
+   * real-time signal called from interrupt or idle task, return with
+   * -EAGAIN.
+   */
+
+  if (sq_empty(&g_sigpendingsignal) || sq_empty(&g_sigpendingaction))
```

Comment:
let's pass pointer to nxsig_alloc_dyn_pending and return error code to avoid checking the preallocated list again.

Reply:
looks good, just several minor comment.